### PR TITLE
Workflow to check PR

### PR DESCRIPTION
**Pull request by liya2017**
Will run this workflow when Pull Request open, synchronize, reopened. Please review the PR https://github.com/nervosnetwork/ckb/pull/2686 before this PR.

Bsed on PR https://github.com/liya2017/ckb/pull/5

TODO:

- [ ]   Add Github Actions self-hosted runner 
- [ ]   Add secrets, then retest and refine the related steps
- [ ] Move mac runner to M1, now mac runner is mini due to known issue.

### CODE REVIEW

**Comment by nervos-bot-user** [comment]
Can one of the admins verify this patch?

**Comment by nervos-bot-user** [comment]
Can one of the admins verify this patch?

File: .github/workflows/pull_request_verify.yaml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+          - os: Linux-runner
+            script_run: make test
+          - os:  mac-runner
+            script_run: make test
+          - os: win-runner
+            script_run: devtools/windows/make test
+            CI: true
+            ImageOS: 'windows-2019'
+            BUILD_BUILDID: ${{ github.sha }}
+    needs: checkout
```
Comment (keroro520) [inline .github/workflows/pull_request_verify.yaml:39]: Notice that you depend on the "checkout" job by `needs: checkout`. 
Could we just check out the repository on the first step of a job by using `- uses: actions/checkout@v2`? And what are the differences?

```
env:
  CI: ${{ matrix.CI }}
  ImageOS:  ${{ matrix.ImageOS }}
steps:
  - uses: actions/checkout@v2
  - run: ${{ matrix.script_run }}
```

Comment (liya2017) [reply .github/workflows/pull_request_verify.yaml:39]: My opinion is define the job name and check logs clearly.
Let me have a try, will modify it if working check out the repository on the first step of a job by using - uses: actions/checkout@v2.

File: .github/workflows/pull_request_verify.yaml [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        include:
+          - os: Linux-runner
+            script_run: make integration
+          - os:  mac-runner
+            script_run: make integration
+          - os: win-runner
+            script_run: devtools/windows/make CKB_TEST_SEC_COEFFICIENT=5 CKB_TEST_ARGS="-c 4 --no-report" integration
+            CI: true
+            SENTRY_DSN: "https://15373165fbf2439b99ba46684dfbcb12@sentry.nervos.org/7"
+    needs: UnitTest
```
Comment (doitian) [inline .github/workflows/pull_request_verify.yaml:52]: These jobs can run in parallel. `Integration_Test` does not need to wait for `UnitTest`.

File: .github/workflows/pull_request_verify.yaml [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    env:
+      CI: ${{ matrix.CI }}
+      SENTRY_DSN: ${{ matrix.SENTRY_DSN }}
+
+  Linters:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [Linux-runner, mac-runner]
+    needs: Integration_Test
```
Comment (doitian) [inline .github/workflows/pull_request_verify.yaml:66]: These jobs can run in parallel. `Linters` does not need to wait for `Integration_Test`.

File: .github/workflows/pull_request_verify.yaml (hunk 2 above)
Comment (liya2017) [reply .github/workflows/pull_request_verify.yaml]: Modified it, the jobs are queued for processing although these jobs can run in parallel.

File: .github/workflows/pull_request_verify.yaml (hunk 3 above)
Comment (liya2017) [reply .github/workflows/pull_request_verify.yaml]: modified

**Review by quake — APPROVED** [review]

**Pull request merged**